### PR TITLE
Add an --api flag to generate project without view layer

### PR DESCRIPTION
**Pull request by CrowdHailer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ $ mix raxx.new my_app
 
 ### Options
 
+- `api`: TODO
```
Comment (CrowdHailer) [inline README.md]: Finish this documentation

**Comment by nietaki** [comment]
Looks sensible. But I'm wondering if it's worth it to go a step further: Instead of having an `--api` flag that changes the generated project from web-only to api-only, having two flags that allow you to pick if you want the web and the api parts individually. Those could be something like `--api` and `--no-web` and you could have just api, just web, or both, choosing neither would error out with a helpful message.

I think that would nicely showcase router sections and separating things in the Raxx context. 

Do you think that would be worth the work? Do you see anything that would make this hard to do?

**Comment by CrowdHailer** [comment]
I didn't really think about doing it that way. I was trying to optimise for the API only approach, I want to write an article about building API's. 

It don't see any reason why it couldn't be done. There is a few things that might make it more work.
i.e. if you have both is the api namespaced under `/api` and not if only `--api` is chosen. Also you have the chose of splitting at the router level or alternatively starting two ace services on different ports one web one api

**Comment by nietaki** [comment]
The reason why I'm suggesting that is because I feel like it would hit a very common use-case, where you want to make an API with a simple javascript-driven interface for it. I feel like it would make Raxx an even better platform for quickly build ing tech demos, for example.

With the "api or website" approach, if I was to make such a project, I'd probably have raxx_kit generate a website template for me (since it requires more work to set up on your own) and make the api part myself.

> i.e. if you have both is the api namespaced under `/api` and not if only `--api` is chosen. 

I think having those always namespaced wouldn't be the worst thing.

> Also you have the chose of splitting at the router level or alternatively starting two ace services on different ports one web one api

I see what you mean. I think having one service on the same port would be easier to wrap one's head around, showcase sections and splitting stuff on the router level, and not make it seem like you need to work on two separate ports if you want to have a website and an api.

I don't know if you feel like that's idiomatic enough but I think both the service (What is now `Demo.API` or `Demo.Web`) and the router could both have consistent names outside of the namespaces, just inside `lib/demo`. You could argue that each of the "contexts" should have their own router the top-level one should just dispatch to, but it feels like kind of an advanced move and is something that's very easy to move to on your own if you do want to do it - for the freshly generated project I think one router with sections is plenty good.

Sorry if I'm more opinionated about it than I need to be, I'm just trying to put myself in the shoes of a potential user (which I think I will be soon, with some of my Hoplon experiments).

**Comment by CrowdHailer** [comment]
> Sorry if I'm more opinionated about it than I need to be, I'm just trying to put myself in the shoes of a potential user (which I think I will be soon, with some of my Hoplon experiments).

No it all makes sense.
Though I'm wondering would you have one router. in which case it needs to live outside the API or WWW namespace?

Mulling, though I am tempted to release as is and open an issue to be able to use both at the same time. That would be useful for what I am playing with now and make the path forward clear

**Comment by nietaki** [comment]
> Though I'm wondering would you have one router. in which case it needs to live outside the API or WWW namespace?

Yes, that was my suggestion, for the ease of development of `raxx_kit` and the ease of onboarding to the new project, even if things are not as separated as they could have otherwise been. I like it as a compromise.

I'm definitely ok with releasing this as is, it just seemed like the amount of change to get there directly wasn't too much and changing the semantics of new flags so frequently wasn't necessary.

**Comment by CrowdHailer** [comment]
Decided to release as is, note on flag semantics I think that actually releasing as is and then upgrading can be done without any change to semantics.

If no flag given, it's equivalent to using `--www` (not that we have that option yet)
Using `--api` stays as it is.
In a follow up version it should be possible to use `--www --api` and get both parts of the scaffolding

**Pull request merged**